controller/user-controller.js

```diff
@@ -0,0 +1,41 @@
+const {userService} = require('../services/user-services');
+
+const signUp = async (req, res) => {
+  try {
+    const {email, password, firstname, lastname, skintype} =req.body;
+
+    await userService.signUp(email, password, firstname, lastname, skintype);
+
+    if (!email|| !password || !firstname || !lastname){
+      throw new Error("KEY Error!");
+    }
```

Comment:
두 코드의 순서를 바꿔주세요.

service레이어의 함수의 인자로 데이터를 전달하기 전에 해당 데이터들이 request 객체에 제대로 포함되어 있는지 확인하는 코드가 우선되어야 합니다.

```suggestion
    if (!email|| !password || !firstname || !lastname){
      throw new Error("KEY Error!");
    };
    
    await userService.signUp(email, password, firstname, lastname, skintype);
```